src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -312,6 +331,16 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
+            if (ProvideCommandLineInvocation)
+            {
+                CommandLineInvocation = GenerateResponseFileContents();
```

Comment:
I love the simplicity of this. I worry that in "real" projects it may be a surprisingly large string that might have sub-optimal GC handling. Any sense if something like this has been done previously in build/project system world?